modules/Kconfig.cmsis

```diff
@@ -34,7 +34,18 @@ menuconfig CMSIS_NN
 	depends on CPU_CORTEX
 	depends on NEWLIB_LIBC
 	select CMSIS_DSP
-
+	select CMSIS_DSP_BASICMATH
+	select CMSIS_DSP_COMPLEXMATH
+	select CMSIS_DSP_CONTROLLER
+	select CMSIS_DSP_FASTMATH
+	select CMSIS_DSP_FILTERING
+	select CMSIS_DSP_MATRIX
+	select CMSIS_DSP_STATISTICS
+	select CMSIS_DSP_SUPPORT
+	select CMSIS_DSP_TRANSFORM
+	select CMSIS_DSP_SVM
+	select CMSIS_DSP_BAYES
+	select CMSIS_DSP_DISTANCE
```

Comment:
Pretty sure not all these are required.

Reply:
Do you know of an easy way to find out which ones are used?

Reply:
I am afraid the "trial-and-error approach" would be the quickest/easiest in this case.

Reply:
I've made `grep -rn './' -e 'function_name'` for all functions listened on https://arm-software.github.io/CMSIS_5/DSP/html/modules.html in the CMSIS/NN folder.

None of all them are used - can this be correct or am I missing something?

Reply:
After further investigation I assume, that they only need the Header `arm_math.h` for type definitions. I'll therefore set all CMSIS_DSP components, except CMSIS_DSP itself, to n

Reply:
> After further investigation I assume, that they only need the Header `arm_math.h` for type definitions. I'll therefore set all CMSIS_DSP components, except CMSIS_DSP itself, to n

I did a quick scan of the CMSIS-NN source code and that does seem to be the case. We will see what kind of errors we get (if any) when we build the CMSIS-NN tests in the future.